src/postamble.js

```diff
@@ -188,19 +188,12 @@ Module['callMain'] = function callMain(args) {
   ensureInitRuntime();
 
   var argc = args.length+1;
-  function pad() {
-    for (var i = 0; i < {{{ QUANTUM_SIZE }}}-1; i++) {
-      argv.push(0);
-    }
-  }
-  var argv = [allocate(intArrayFromString(Module['thisProgram']), 'i8', ALLOC_NORMAL) ];
-  pad();
-  for (var i = 0; i < argc-1; i = i + 1) {
-    argv.push(allocate(intArrayFromString(args[i]), 'i8', ALLOC_NORMAL));
-    pad();
+  var argv = _malloc(argc * {{{ Runtime.POINTER_SIZE }}});
```

Comment:
Should probably be `_malloc((argc + 1) * ...)` because of `argv[argc]`.

Reply:
Good catch, thanks.

I guess this didn't break anything since this is the first malloc, and so the write didn't trample anything...